src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java

```diff
@@ -488,22 +495,241 @@ private void testFileWithoutInfoLineInHeader(final String name, final Class<? ex
+        final Predicate<VariantContext> filter = vc -> vc.getAttributeAsInt("AC", -1) > 0 ||
+                vc.getAttributeAsString("SVTYPE", "").equals("DEL");
+
+        final Pair<VCFHeader, List<VariantContext>> preFilter = VariantContextTestUtils.readEntireVCFIntoMemory(svHGDBMultiSampleVcf.toString());
+        final int expectedCount = (int) preFilter.getRight().stream().filter(filter).count();
+
+        runCommandLine(Arrays.asList("-V", svHGDBMultiSampleVcf.toString(),
+                        "-O", testOutput.getAbsolutePath(),
+                        "--select", "SVTYPE == 'DEL'",
+                        "--select", "AC > 0"),
+                SelectVariants.class.getSimpleName());
+        final Pair<VCFHeader, List<VariantContext>> result = VariantContextTestUtils.readEntireVCFIntoMemory(testOutput.getAbsolutePath());
+        final int actualCount = result.getRight().size();
+        Assert.assertEquals(expectedCount, actualCount);
+    }
+
+    // Test JEXL expressions like "AC > 0 && AF > 0.001"
+    // The recommended usage for logical-or is to place individual expressions as separate --select command
+    // e.g. -select AC > 0 -select AF > 0.001 for "AC > 0 || AF > 0.001"
+    @Test
+    public void testLogicalAndOrInJexl() {
+        beforeJexlTest();
+        final File testOutput = createTempFile("multiple_jexl_test", "vcf");
+
+        final Predicate<VariantContext> andFilter = vc -> vc.getAttributeAsInt("AC", -1) > 0 &&
+                vc.getAttributeAsString("SVTYPE", "").equals("DEL");
+        final Pair<VCFHeader, List<VariantContext>> preFilter = VariantContextTestUtils.readEntireVCFIntoMemory(svHGDBMultiSampleVcf.toString());
+        final int expectedCount = (int) preFilter.getRight().stream().filter(andFilter).count();
+
+        runCommandLine(Arrays.asList("-V", svHGDBMultiSampleVcf.toString(),
+                        "-O", testOutput.getAbsolutePath(),
+                        "--select", "SVTYPE == 'DEL' && AC > 0"),
+                SelectVariants.class.getSimpleName());
+        final Pair<VCFHeader, List<VariantContext>> result = VariantContextTestUtils.readEntireVCFIntoMemory(testOutput.getAbsolutePath());
+        final int actualCount = result.getRight().size();
+        Assert.assertEquals(expectedCount, actualCount);
+    }
+
+
+    /** One of the five variants does not have the GQ field. It is silently removed from the output. **/
+    @Test
+    public void testJexlGenotypeFilter() {
+        final File testOutput = createTempFile("jexl_genotype_test", "vcf");
+        beforeJexlTest();
+        final int threshold = 1;
+        final Predicate<VariantContext> filter = vc -> vc.getGenotypes().stream().anyMatch(g -> g.getGQ() > threshold);
+        final Pair<VCFHeader, List<VariantContext>> preFilter = VariantContextTestUtils.readEntireVCFIntoMemory(svHGDBMultiSampleVcf.toString());
+        final int expectedNumVCs = (int) preFilter.getRight().stream().filter(filter).count();
```

Comment:
In these new tests you added, have you confirmed in each case that the number of expected records is > 0? Otherwise the tests will not be as meaningful as they could be.

Reply:
Added extra assert statements to check this.

Reply:
Done